docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,673 @@
+
+An Ion Schema MAY include extra top-level values that are not explicitly specified in the Ion Schema specification, but any top-level open content MUST NOT be annotated with a *reserved symbol*.
+Note that Ion Schema version markers are always interpreted as Ion Schema version markers and can never be valid open content.
+
+No other open content is allowed except for what is explicitly identified in this section.
+
+### Reserved Symbols
+
+The set of reserved symbols SHALL be all symbols matching the regular expression `^(\$ion_schema(_.*)?|[a-z][a-z0-9]*(_[a-z0-9]+)*)$`.
+Informally stated, this is the symbol `$ion_schema`, all symbols starting with `$ion_schema_`, and all [identifier symbols](https://amzn.github.io/ion-docs/docs/spec.html#symbol) that are *snake case* and start with an unaccented ascii, lower-case letter.
+
+### Keywords
+A keyword is a reserved symbol that has been assigned a meaning by the Ion Schema specification.
+Whether a reserved symbol is considered a keyword is context dependent.
+* Within the schema header, the keywords SHALL be `imports` and `user_content`.
+* Within a type definition, the keywords SHALL be `name`, `occurs`, and `id` , as well as all the constraints defined in this version of the Ion Schema specification (which are `all_of`, `annotations`, `any_of`, `byte_length`, `codepoint_length`, `container_length`, `contains`, `element`, `exponent`, `field_names`, `fields`, `id`, `name`, `not`, `occurs`, `one_of`, `ordered_elements`, `precision`, `regex`, `timestamp_offset`, `timestamp_precision`, `type`, `utf8_byte_length`, and `valid_values`).
+* There are no keywords in a schema footer.
+
+{% capture sample_code %}
+schema_header::{
+  _info: "This schema is about penguins."
+}
+type::{
+  name: adelie,
+  type: penguin,
+  _region: antarctica,
+  _crested: false,
+  _banded: false,
+}
+type::{
+  name: humboldt,
+  type: penguin,
+  _region: south_america,
+  _crested: false,
+  _banded: true,
+}
+schema_footer::{}
+{% endcapture %}
+{% include example.md title="A schema with open content using un-reserved symbols as open content field names." code=sample_code %}
+
+{% capture sample_code %}
+schema_header::{
+  info: "This schema is about penguins."
+  user_content: {
+    schema_header: [info],
+    type: [region, crested, banded],
+  }
+}
+type::{
+  name: adelie,
+  type: penguin,
+  region: antarctica,
+  crested: false,
+  banded: false,
+}
+type::{
+  name: humboldt,
+  type: penguin,
+  region: south_america,
+  crested: false,
+  banded: true,
+}
+schema_footer::{}
+{% endcapture %}
+{% include example.md title="The same schema with open content using reserved symbols as open content field names." code=sample_code %}
+
+# Compatibility with Ion Schema 1.0
+
+Ion Schema 2.0 will be fully interoperable with Ion Schema 1.0.
+
+* Any Ion Schema implementation with support for Ion Schema 2.0 must also support reading Ion Schema 1.0.
```

Comment:
We need to have a document that spells out the parts of the spec are different in 2.0 so an implementer doesn't have to manually cross reference the 1.0 and 2.0 specs to discover the changes. Maybe the RFC suffices for this? Whichever doc we pick, this would be a good place to link to it to highlight that the differences are manageable.

Reply:
Created #92